src/D2L.Security.OAuth2/Keys/KeyManagementService.cs

```diff
@@ -51,7 +65,9 @@ OAuth2Configuration config
 
 			var now = m_clock.UtcNow;
 
-			if ( current == null || current.ValidTo <= now ) {
+			if ( current == null
+				|| ExpectedTimeOfNewUsableKey( current ) + GetSigningCredentialsRefreshGracePeriod < now

```

Comment:
The implication here is that once this period is passed every call to `GetSigningCredentials` is going to fight to update that key.

That basically matches how it worked before the background refresh anyway. The only things without background refresh for us are Lambda functions though, which are single threaded anyway.